endpoint.go

```diff
@@ -417,36 +297,30 @@ func (ep *endpoint) Join(containerID string, options ...EndpointOption) error {
 		}
 	}()
 
-	err = ep.buildHostsFiles()
+	err = sb.buildHostsFiles(ep)
```

Comment:
Actually here too, buildHostsFiles need to know to which network we are connecting the sandbox so that it retrieves the SrvcRecords from it and need to know the endpoint to get the IP of the interface.

Seems like this operation cannot be completely moved to sandbox creation.